src/models/athleteActivity.ts

```diff
@@ -1,109 +1,62 @@
-  id: bigint;
-  external_id: string;
-  upload_id: bigint;
-  athlete: MetaAthlete;
-  name: string;
-  distance: number;
-  moving_time: number;
-  elapsed_time: number;
-  total_elevation_gain: number;
-  elev_high: number;
-  elev_low: number;
-  type: SportType;
-  sport_type: SportType;
-  start_date: Date;
-  timezone: string;
-  start_latlng: LatLng;
-  end_latlng: LatLng;
-  achievement_count: number;
-  kudos_count: number;
-  comment_count: number;
-  athlete_count: number;
-  photo_count: number;
-  total_photo_count: number;
-  map: PolylineMap;
-  trainer: boolean;
-  commute: boolean;
-  manual: boolean;
-  private: boolean;
-  flagged: boolean;
-  workout_type?: number;
-  upload_id_str: string;
-  average_speed: number;
-  max_speed: number;
-  has_kudoed: boolean;
-  hide_from_home?: boolean;
-  gear_id?: string;
-  kilojoules?: number;
-  average_watts?: number;
-  device_watts?: boolean;
-  max_watts?: number;
-  weighted_average_watts?: number;
-  resource_state: number;
-  start_date_local: Date;
-}
-
-export const activityAthleteSchema = Joi.object<AthleteActivity>({
-  id: Joi.number().required(),
-  external_id: Joi.string().required(),
-  upload_id: Joi.number().required(),
-  athlete: metaAthleteSchema.required(),
-  name: Joi.string().required(),
-  distance: Joi.number().required(),
-  moving_time: Joi.number().required(),
-  elapsed_time: Joi.number().required(),
-  total_elevation_gain: Joi.number().required(),
-  elev_high: Joi.number().required(),
-  elev_low: Joi.number().required(),
-  type: sportTypeSchema.required(),
-  sport_type: sportTypeSchema.required(),
-  start_date: Joi.date().required(), //NOTE: dubbio su questo DATE
-  timezone: Joi.string().required(),
-  start_latlng: latLngSchema.required(),
-  end_latlng: latLngSchema.required(),
-  achievement_count: Joi.number().required(),
-  kudos_count: Joi.number().required(),
-  comment_count: Joi.number().required(),
-  athlete_count: Joi.number().required(),
-  photo_count: Joi.number().required(),
-  total_photo_count: Joi.number().required(),
-  map: polylineMapSchema.required(),
-  trainer: Joi.boolean().required(),
-  commute: Joi.boolean().required(),
-  manual: Joi.boolean().required(),
-  private: Joi.boolean().required(),
-  flagged: Joi.boolean().required(),
-  workout_type: Joi.number().allow(null),
-  upload_id_str: Joi.string().required(),
-  average_speed: Joi.number().required(),
-  max_speed: Joi.number().required(),
-  has_kudoed: Joi.boolean().required(),
-  hide_from_home: Joi.boolean(),
-  gear_id: Joi.string().required().allow(null),
-  kilojoules: Joi.number(),
-  average_watts: Joi.number(),
-  device_watts: Joi.boolean(),
-  max_watts: Joi.number(),
-  weighted_average_watts: Joi.number(),
-  resource_state: Joi.number().required(),
-  start_date_local: Joi.date().required(),
+export const ActivityAthleteSchema = z.object({
+  id: z.coerce.bigint(),
+  external_id: z.string(),
+  upload_id: z.coerce.bigint(),
```

Comment:
Usiamo direttamente bigint
```suggestion
  upload_id: z.bigint(),
```

Reply:
Qua avevo lasciato ".coerce" perché con "bigint" sembra che Zod lo convalida come "number" lo stesso. 

  {
    "code": "invalid_type",
    "expected": "bigint",
    "received": "number",
    "path": [
      "id"
    ],
    "message": "Expected bigint, received number"
  },
  {
    "code": "invalid_type",
    "expected": "bigint",
    "received": "number",
    "path": [
      "upload_id"
    ],
    "message": "Expected bigint, received number"
  }

Reply:
Ho cercato più informazioni su 'coerce', se non ho capito male quello che fa è convertire il campo nel tipo definito dallo schema.

Per ora ci va bene così, poi ci ritorneremo magari.